File: 066/main.js
/**
 * @param {number[]} digits
 * @return {number[]}
 */
var plusOne = function (digits) {
  const len = digits.length
  let flag = true
  for (let i = len - 1; i >= 0; i--) {
    if (digits[i] === 9 && flag) {
      digits[i] = 0
    } else if (flag) {
      digits[i]++
      flag = false
    } else {
      flag = false
      break
    }
  }
  if (flag) digits.unshift(1)
  return digits
}

console.log(plusOne([9, 9, 9]))
console.log(plusOne([0]))

File: 066/main.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        int len = digits.size();
        int flag = 1;
        for (int i = len - 1; i >= 0; i--) {
            if (digits[i] == 9 && flag) {
                digits[i] = 0;
            } else if (flag) {
                digits[i]++;
                flag = 0;
            } else {
                flag = 0;
                break;
            }
        }
        if (flag) digits.insert(digits.begin(), 1);
        return digits;
    }
};

int main()
{
    Solution sol;
    vector<int> digits;
    digits.push_back(9);
    digits.push_back(9);
    digits.push_back(9);
    vector<int> res = sol.plusOne(digits);
    for (int i = 0; i < res.size(); i++) {
        cout << res[i] << " ";
    }
    return 0;
}
